docs/components/SelectView.jsx

```diff
@@ -199,8 +216,21 @@ export default class SelectView extends Component {
             },
             {
               name: "options",
-              type: "Array",
-              description: "Possible options, must contain objects with label and value attributes.",
+              type: "{label: string; value: string}[]",
+              description: "Possible options, must contain objects with label and value attributes. Must not be set if `lazy`.",
+              optional: true,
+            },
+            {
+              name: "lazy",
+              type: "boolean",
+              description: "Set to true to fetch options asynchronously using the `loadOptions` function.",
+              defaultValue: "false",
+              optional: true,
+            },
+            {
+              name: "loadOptions",
+              type: "(input: string) => Promise<{options: {label: string; value: string}[]}>",
+              description: "Function to load options for a given search input. Required if `lazy`.",
```

Comment:
There's some nuance to the way react-select does this that we should probably expose, like how unless we supply `autoload=false` to the React Select component it will default the data to whatever `loadOptions('')` returns; and that options for previously searched inputs are cached (unless explicitly set not to in the React Select component). At a minimum, this behavior ought to be documented, and at a maximum we should consider if this sort of stuff are things we'd like to make configurable, perhaps via a passthrough `reactSelectOptions` property.